pokt-network/pocket
# [Telemetry]: Add Telemetry Infrastructure + Module Interface + Prometheus Implementation

## Description
##### PR 1/3 for `milestone/v1-telemetry`.


1. Telemetry interface, module and required infrastructure tooling to support it


Notice that we are merging against `milestone/v1-telemetry`. When all telemetry PRs are merged, we will merge `milestone/v1-telemetry` into main.

## Changes

- [x] Added telemetry services to docker-compose
   - [x] loki
   - [x] grafana
   - [x] victoria-metrics
- [x] Defined a telemetry module interface
- [x] Implemented the telemetry module interface for Prometheus
- [x] Implemented a NOOP telemetry module for when telemetry is not desired.

## How to test?

1. Run your stack
```
$ make compose_and_watch
```

3. Check if the infrastructure is up by opening:
  1. [Grafana locally](http://localhost:3000/)
  2. [Victoria Metrics UI](http://localhost:8428/vmui/)
  4. Check to see if Victoria Metrics detects the prometheus endpoint by checking [targets](http://localhost:8428/targets). You should see a section named: `pocket-nodes (4/4)`, with the prometheus endpoint details and status underneath.

  3. Loki by navigating Grafana to the 'Explore' section, and running the query `{host="desktop-docker"} | logfmt` to see logs popping up.

## Documentation

Documentation is available at the telemetry module's [README.md](https://github.com/pokt-network/pocket/blob/telemetry/add-telemetry-infrastructure/shared/modules/telemetry/README.md)

## Owners
@okdas 
@derrandz
